reference/reference.go

```diff
@@ -168,6 +169,20 @@ func ParseIDOrReference(idOrRef string) (digest.Digest, Named, error) {
 	return "", ref, err
 }
 
+// Match reports whether ref matches the specified pattern
+func Match(pattern string, ref Named) bool {
+	// FIXME: should we support pattern for NamedTagged or not ?
+	patternRef, err := ParseNamed(pattern)
+	if _, ok := patternRef.(NamedTagged); err == nil && ok {
+		return ref.String() == pattern
+	}
+	matched, err := path.Match(pattern, ref.Name())
```

Comment:
If we want to support globbing over any of the reference types, yes we should do that.